### PR TITLE
feat(Modal): promoted next version

### LINKED ISSUE #8078: Modal: Deprecate modal
Deprecate the modal

- Move the Modal code to the deprecated directory (`react-core\src\deprecated\`). The documentation would show up under a "React deprecated" tab on the Modal page  of the website.
- Promote the "next Modal" to our default Modal component by moving it to `react-core\src\components\`.  

related to https://github.com/patternfly/patternfly-react/issues/8076

### PR DESCRIPTION
**Pull request by thatblindgeye**
**What**: Closes #8078 

**Additional issues**:

### CODE REVIEW

**Review by wise-king-sullyman — APPROVED** [review]

**Review by kmcfaul — APPROVED** [review]

File: packages/react-core/src/components/AboutModal/AboutModal.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { AboutModalBoxHeader } from './AboutModalBoxHeader';
 import { AboutModalBoxBrand } from './AboutModalBoxBrand';
 import { AboutModalBoxCloseButton } from './AboutModalBoxCloseButton';
 import { AboutModalBox } from './AboutModalBox';
-import { Modal, ModalVariant } from '../Modal';
```
Comment (tlabaj) [inline packages/react-core/src/components/AboutModal/AboutModal.tsx:9]: we should be using the new modal here.  I am ok with follow up issue to do that, but it should be done for Beta.

File: packages/react-core/src/components/Label/examples/LabelGroupEditableAddModal.tsx [hunk 2]
```diff
@@ -16,6 +14,7 @@ import {
   Radio,
   Popper
 } from '@patternfly/react-core';
+import { Modal as ModalDeprecated, ModalVariant as ModalVariantDeprecated } from '@patternfly/react-core/deprecated';
```
Comment (tlabaj) [inline packages/react-core/src/components/Label/examples/LabelGroupEditableAddModal.tsx:17]: same comment as above here

File: packages/react-core/src/components/Wizard/examples/WizardWithinModal.tsx [hunk 3]
```diff
@@ -8,13 +9,13 @@ export const WizardWithinModal = () => {
     <>
       <Button onClick={() => setIsModalOpen(true)}>Show Modal</Button>
 
-      <Modal
+      <ModalDeprecated
```
Comment (tlabaj) [inline packages/react-core/src/components/Wizard/examples/WizardWithinModal.tsx:12]: we should use new modal here.

**Review by tlabaj — CHANGES_REQUESTED** [review]
General comment.  We should not be using the deprecated modal in our components or examples/demos. We can open a follow up issue for that.

**Comment by thatblindgeye** [comment]
In regards to above, followup created for updating examples: https://github.com/patternfly/patternfly-react/issues/10363

**Review by tlabaj — APPROVED** [review]

**Pull request merged**